index.js

```diff
@@ -38,14 +41,31 @@ function loadModule(config) {
  */
 function initializeBookend(defaultModules, list) {
     return list.map((m) => {
-        if (typeof m === 'string' && defaultModules[m]) {
+        let name;
+        let alias;
+
+        if (typeof m === 'string') {
+            name = m;
+            alias = m;
+        } else {
+            // Use the first key as a module name if name is not set
+            // to support the format: { moduleName: moduleAlias }
+            name = m.name || Object.keys(m)[0];
+            alias = m[name] || name;
+        }
+
+        if (defaultModules[name]) {
             return {
-                obj: defaultModules[m],
-                name: m
+                obj: defaultModules[name],
+                name: alias
             };
         }
 
-        return loadModule(m);
+        return loadModule({
+            name,
```

Comment:
why can't we have:
``` 
loadModule({
  name: alias, 
  config: m.config || {}
});
```
Then we won't need to make changes to `loadModule`

Reply:
If we pass an alias to `loadModule`, as `name`, `loadModule` will try to load the module of `alias` instead of `name` and an error will occur. 
So, `loadModule` should be called with both name and alias, load a module specified by `name` and return an object with `name: alias`.

Reply:
I see. Makes sense! Thanks for explaining @catto  :)